File: 프로그래머스/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];

        Map<String, Integer> map = new HashMap<>();
        
        for(int i = 0; i < words.length; i++){
            String cur = words[i];
            
            if(i>0){
                String prev = words[i-1];
                
                char last = prev.charAt(prev.length()-1);
                char first = cur.charAt(0);
                
                if(map.containsKey(cur) || last != first){
                    answer[0] = i % n + 1;
                    answer[1] = i / n + 1;
                    break;
                }
            }
            map.put(cur,1);
        }
        
        return answer;
    }
}

File: 프로그래머스/영어 끝말잇기.py
def solution(n, words):
    answer = [0, 0]
    dup = []
    dup.append(words[0])
    
    for i in range(1, len(words)):
        cur = words[i]        
        prev = words[i-1]
        
        last = prev[-1]
        first = cur[0]
            
        if cur in dup or last != first:
            answer = [i%n+1, i//n+1]
            break
        
        dup.append(cur)
    
    return answer